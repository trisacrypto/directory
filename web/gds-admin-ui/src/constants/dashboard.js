export const Status = {
    SUBMITTED: "SUBMITTED",
    EMAIL_VERIFIED: "EMAIL_VERIFIED",
    PENDING_REVIEW: "PENDING_REVIEW",
    REVIEWED: "REVIEWED",
    ISSUING_CERTIFICATE: "ISSUING_CERTIFICATE",
    VERIFIED: "VERIFIED",
    REJECTED: "REJECTED",
    APPEALED: "APPEALED",
    ERRORED: "ERRORED",
    NO_VERIFICATION: "NO_VERIFICATION"
}

export const StatusLabel = {
    SUBMITTED: "Submitted",
    EMAIL_VERIFIED: "Email Verified",
    PENDING_REVIEW: "Pending Review",
    REVIEWED: "Reviewed",
    ISSUING_CERTIFICATE: "Issuing Certificate",
    VERIFIED: "Verified",
    REJECTED: "Rejected",
    APPEALED: "Appealed",
    ERRORED: "Errored",
    NO_VERIFICATION: "No Verification"
}