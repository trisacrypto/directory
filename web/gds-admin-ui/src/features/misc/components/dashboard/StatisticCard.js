import PropTypes from 'prop-types';
import { Card } from 'react-bootstrap';

function StatisticCard({ count, title, icon }) {
  return (
    <Card className="shadow-none m-0">
      <Card.Body className="text-center">
        {icon}
        <h3>{count || 0}</h3>
        <p className="text-muted font-15 mb-0">{title}</p>
      </Card.Body>
    </Card>
  );
}

StatisticCard.propTypes = {
  count: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  icon: PropTypes.node.isRequired,
};

export default StatisticCard;
