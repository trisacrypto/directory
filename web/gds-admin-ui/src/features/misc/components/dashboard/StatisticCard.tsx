import { ReactNode } from 'react';
import { Card } from 'react-bootstrap';

type StatisticCardProps = {
    count: number;
    title: string;
    icon: ReactNode;
};

function StatisticCard({ count, title, icon }: StatisticCardProps) {
    return (
        <Card className="shadow-none m-0">
            <Card.Body className="text-center">
                {icon}
                <h3>{count || 0}</h3>
                <p className="text-muted font-15 mb-0">{title}</p>
            </Card.Body>
        </Card>
    );
}

export default StatisticCard;
