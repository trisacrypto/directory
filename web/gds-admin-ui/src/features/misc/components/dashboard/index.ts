export {default as StatisticCard} from './StatisticCard'
export {default as Statistics} from './Statistics'
export {default as Status} from './Status'
export {default as PendingAndRecentActivity} from './PendingAndRecentActivity'
export {default as TasksChart} from './TasksChart'
export {default as VaspsByCountryChart} from './VaspsByCountryChart'