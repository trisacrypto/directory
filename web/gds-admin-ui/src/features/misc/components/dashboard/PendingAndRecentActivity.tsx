import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import React from 'react';
import { Card, Dropdown, Table } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';
import SimpleBar from 'simplebar-react';

import CiphertraceFavicon from '@/assets/images/ciphertrace.ico';
import TrisaFavicon from '@/assets/images/trisa_favicon.png';
import NoData from '@/components/NoData';
import { StatusLabel } from '@/constants/index';
import { actionType, useModal } from '@/contexts/modal';
import { useGetVasps } from '@/features/vasps';

dayjs.extend(relativeTime);

const PENDING_REVIEW_QUERY_PARAMS = 'status=pending_review';

const PendingReviewsTable = ({ data }: any) => {
    const [vasp, setVasp] = React.useState({ name: '', id: '' });
    const { dispatch } = useModal();
    const history = useHistory();

    const handleResendEmailClick = (name: any) => {
        setVasp(name);
        dispatch({ type: actionType.SEND_EMAIL_MODAL, payload: { vasp } });
    };

    return (
        <SimpleBar style={{ maxHeight: 350 }} className="task">
            <Table responsive className="table table-centered table-nowrap table-hover mb-0 z-index-2">
                <tbody>
                    {data?.map((vasp: any) => (
                        <tr key={vasp.id}>
                            <td
                                onClick={() => history.push(`/vasps/${vasp?.id}`)}
                                className="d-flex gap-2 align-items-center"
                                role="button">
                                <div>
                                    {vasp?.traveler ? (
                                        <img src={CiphertraceFavicon} width="30" alt="CipherTrace" />
                                    ) : (
                                        <img src={TrisaFavicon} width="30" className="img-fluid" alt="TRISA" />
                                    )}
                                </div>
                                <div>
                                    <h5 className="font-14 my-1 gap-2 d-flex">{vasp?.name}</h5>
                                    <span className="text-muted font-13">{dayjs(vasp?.last_updated).fromNow()}</span>
                                </div>
                            </td>
                            <td>
                                <span className="text-muted font-13">Status</span> <br />
                                <span className="badge badge-warning-lighten">
                                    {(StatusLabel as any)[vasp?.verification_status]}
                                </span>
                            </td>
                            <td>
                                <span className="text-muted font-13">Emails</span>
                                <div className="font-14 mt-1 fw-normal">
                                    {vasp?.verified_contacts &&
                                        Object.keys(vasp?.verified_contacts).map((contact) => (
                                            <span
                                                key={contact}
                                                className={`badge ${
                                                    vasp?.verified_contacts[contact]
                                                        ? 'badge-success-lighten'
                                                        : 'badge-danger-lighten'
                                                }`}
                                                style={{ marginRight: 4 }}>
                                                {contact}
                                            </span>
                                        ))}
                                </div>
                            </td>
                            <td className="table-action text-center" style={{ width: '90px' }}>
                                <Dropdown align="end">
                                    <Dropdown.Toggle variant="link" className="arrow-none card-drop p-0 shadow-none">
                                        <i className="mdi mdi-dots-horizontal" />
                                    </Dropdown.Toggle>
                                    <Dropdown.Menu>
                                        <Dropdown.Item onClick={() => history.push(`/vasps/${vasp?.id}`)}>
                                            {' '}
                                            <span className="mdi mdi-eye-outline" /> View
                                        </Dropdown.Item>
                                        <Dropdown.Item onClick={() => handleResendEmailClick(vasp)}>
                                            {' '}
                                            <span className="mdi mdi-email-edit-outline" /> Resend email
                                        </Dropdown.Item>
                                    </Dropdown.Menu>
                                </Dropdown>
                            </td>
                        </tr>
                    ))}
                </tbody>
            </Table>
        </SimpleBar>
    );
};

const PendingAndRecentActivity = () => {
    const { data } = useGetVasps({
        queryParams: PENDING_REVIEW_QUERY_PARAMS,
    });

    return (
        <Card className="w-100">
            <Card.Body>
                <h4 className="header-title mb-3">Pending and Recent Activity</h4>
                {data?.vasps?.length ? (
                    <PendingReviewsTable data={data?.vasps} />
                ) : (
                    <NoData title="No available pending registrations" />
                )}
            </Card.Body>
        </Card>
    );
};

export default PendingAndRecentActivity;
