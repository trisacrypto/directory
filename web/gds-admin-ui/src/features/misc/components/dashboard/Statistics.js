import PropTypes from 'prop-types';
import { Card, Col, Row } from 'react-bootstrap';

import OvalLoader from '@/components/OvalLoader';

import StatisticCard from './StatisticCard';

const Statistics = ({ data }) => (
  <>
    {data ? (
      <Row>
        <Col>
          <Card className="widget-inline">
            <Card.Body className="p-0">
              <Row className="g-0">
                <Col sm={6} xl={3}>
                  <StatisticCard
                    title="All VASPs"
                    count={data?.vasps_count}
                    icon={<i className="dripicons-briefcase text-muted font-24" />}
                  />
                </Col>

                <Col sm={6} xl={3} className="border-start">
                  <StatisticCard
                    title="Pending Registrations"
                    count={data?.pending_registrations}
                    icon={<i className="dripicons-checklist text-muted font-24" />}
                  />
                </Col>

                <Col sm={6} xl={3} className="border-start">
                  <StatisticCard
                    title="Verified Contacts"
                    count={data?.verified_contacts}
                    icon={<i className="dripicons-user-group text-muted font-24" />}
                  />
                </Col>

                <Col sm={6} xl={3} className="border-start">
                  <StatisticCard
                    title="Certificates Issued"
                    count={data?.certificates_issued}
                    icon={<i className="dripicons-copy text-muted font-24" />}
                  />
                </Col>
              </Row>
            </Card.Body>
          </Card>
        </Col>
      </Row>
    ) : (
      <div className="mb-3">
        <OvalLoader />
      </div>
    )}
  </>
);

Statistics.propTypes = {
  data: PropTypes.object,
};

export default Statistics;
