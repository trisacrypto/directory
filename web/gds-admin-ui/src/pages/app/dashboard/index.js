// @flow
import React from 'react';
import { Row, Col } from 'react-bootstrap';
import { useDispatch } from 'react-redux';

// components
import PageTitle from '../../../components/PageTitle';
import { fecthRegistrationsReviews, fetchCertificates, fetchSummary, fetchVasps } from '../../../redux/dashboard/actions';

import Statistics from './Statistics';
import Status from './Status';
import Tasks from './Tasks';
import TasksChart from './TasksChart';


const ProjectDashboardPage = (): React$Element<React$FragmentType> => {
    const dispatch = useDispatch();

    React.useEffect(() => {
        dispatch(fetchCertificates());
        dispatch(fetchVasps());
        dispatch(fetchSummary())
        dispatch(fecthRegistrationsReviews())
    }, [dispatch])

    return (
        <React.Fragment>
            <PageTitle
                breadCrumbItems={[
                    { label: 'Summary', path: '/dashboard', active: true }
                ]}
                title={'Dashboard'}
            />

            <Statistics />

            <Row style={{ height: "500px" }}>
                <Col lg={4}>
                    <Status />
                </Col>
                <Col lg={8} style={{ overflowY: "scroll", height: "100%" }}>
                    <Tasks />
                </Col>
            </Row>
            <Row>
                <Col>
                    <TasksChart />
                </Col>
            </Row>
        </React.Fragment>
    );
};

export default ProjectDashboardPage;
