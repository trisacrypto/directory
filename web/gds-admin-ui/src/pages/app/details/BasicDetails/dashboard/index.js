import React from 'react';
import { Row, Col } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

// components
import PageTitle from 'components/PageTitle';
import { fecthRegistrationsReviews, fetchCertificates, fetchPendingVasps, fetchSummary } from 'redux/dashboard/actions';

import Statistics from './Statistics';
import Status from './Status';
import Tasks from './Tasks';
import TasksChart from './TasksChart';
import { getPendingVaspsData, getPendingVaspsError, getPendingVaspsLoadingState, getSummaryData } from 'redux/selectors/dashboard';
import useSafeDispatch from 'hooks/useSafeDispatch';


const ProjectDashboardPage = () => {
    const _dispatch = useDispatch();
    const safeDispatch = useSafeDispatch(_dispatch)
    const summary = useSelector(getSummaryData)
    const vasps = useSelector(getPendingVaspsData)
    const isVaspsLoading = useSelector(getPendingVaspsLoadingState)
    const pendingVaspsError = useSelector(getPendingVaspsError)

    React.useEffect(() => {
        safeDispatch(fetchCertificates());
        safeDispatch(fetchPendingVasps());
        safeDispatch(fetchSummary())
        safeDispatch(fecthRegistrationsReviews())
    }, [safeDispatch])

    return (
        <React.Fragment>
            <PageTitle
                breadCrumbItems={[
                    { label: 'Summary', path: '/dashboard', active: true }
                ]}
                title={'Dashboard'}
            />

            <Statistics data={summary} />

            <Row>
                <Col lg={4}>
                    <Status statuses={summary?.statuses} />
                </Col>
                <Col sm={12} lg={8} className='d-flex'>
                    <Tasks data={vasps} isLoading={isVaspsLoading} error={pendingVaspsError} />
                </Col>
            </Row>
            <Row>
                <Col>
                    <TasksChart />
                </Col>
            </Row>

        </React.Fragment>
    );
};

export default React.memo(ProjectDashboardPage);
