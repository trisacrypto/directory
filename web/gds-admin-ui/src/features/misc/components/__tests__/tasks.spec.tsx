import { render } from '@/utils/test-utils';
import { PendingAndRecentActivity } from '../dashboard';

describe('<Tasks />', () => {
    it('Should keep the same UI', () => {
        const { container } = render(<PendingAndRecentActivity />);

        //
        expect(container).toMatchSnapshot('pending-registrations');
    });
});
