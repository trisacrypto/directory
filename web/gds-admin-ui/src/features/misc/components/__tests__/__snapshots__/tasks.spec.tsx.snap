// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tasks /> Should keep the same UI: pending-registrations 1`] = `
<div>
  <div
    class="w-100 card"
  >
    <div
      class="card-body"
    >
      <h4
        class="header-title mb-3"
      >
        Pending and Recent Activity
      </h4>
      <div
        class="text-center"
      >
        <div>
          <img
            alt="no data"
            src="no-data-rafiki.svg"
            width="300"
          />
        </div>
        <p>
          No available pending registrations
        </p>
      </div>
    </div>
  </div>
</div>
`;
